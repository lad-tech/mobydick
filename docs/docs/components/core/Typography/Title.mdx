import Title from "./assets/Title.png";

<img src={Title} height={600} />

Reusable text

## Example

```tsx
import {Title} from "@lad-tech/mobydick-core";

export const Example = () => {
  return (
    <Title font={'Primary-H5'}>Example</Title>
  );
};

```
## `Props`

### [`TextInput Props`](https://reactnative.dev/docs/textinput#props)

Inherits [TextInput Props](https://reactnative.dev/docs/textinput#props).

### `font`

| Type                                                                                                  | DEFAULT      |
|:------------------------------------------------------------------------------------------------------|:-------------|
| [TitleProp](https://github.com/lad-tech/mobydick/blob/main/packages/core/src/typography/types.ts#L49) | 'Primary-H1' |

Font for text
