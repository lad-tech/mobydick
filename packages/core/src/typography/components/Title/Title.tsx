import {FC} from 'react';

import {useFontHeader} from '../../hooks/useFontHeader';
import {IHeaderProps} from '../../types';
import Text from '../../../basic/components/Text/Text';

export const Title: FC<IHeaderProps> = ({
  children,
  font = 'Primary-H1',
  style,
  ...props
}) => {
  const {fontStyle} = useFontHeader(font);
  return (
    <Text style={[fontStyle, style]} {...props}>
      {children}
    </Text>
  );
};
