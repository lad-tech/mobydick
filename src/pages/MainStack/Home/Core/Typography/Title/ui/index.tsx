import {FlatListProps, ListRenderItem} from 'react-native';

import {
  defaultTextLightColor,
  FlatList,
  TFontColor,
  TFontHeaderSize,
  Title,
  TitleProp,
  TypographyLegacyProp,
  useStyles,
  View,
} from '@/shared/ui';
import getScreenStyles from '@/shared/styles/getScreenStyles';

const getAllTitle = (): TitleProp[] => {
  const sizes: TFontHeaderSize[] = Object.keys(
    TFontHeaderSize,
  ) as unknown as TFontHeaderSize[];
  const colors: TFontColor[] = Object.keys(defaultTextLightColor).map(name =>
    name.slice(4, name.length),
  ) as unknown as TFontColor[];

  const result: TitleProp[] = [];
  colors.forEach(color => {
    sizes.forEach(size => {
      result.push(`${color}-${size}`);
    });
  });

  return result;
};

const renderItem: ListRenderItem<TitleProp> = ({item}) => (
  <Title font={item}>{item}</Title>
);

const keyExtractor: FlatListProps<TypographyLegacyProp>['keyExtractor'] =
  item => item;
const TitleScreen = () => {
  const [styles] = useStyles(getScreenStyles);
  const data = getAllTitle();

  return (
    <View style={styles.container}>
      <FlatList
        data={data}
        renderItem={renderItem}
        keyExtractor={keyExtractor}
        removeClippedSubviews={true}
      />
    </View>
  );
};

export default TitleScreen;
