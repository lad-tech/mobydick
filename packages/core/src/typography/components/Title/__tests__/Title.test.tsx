import {render} from '@testing-library/react-native';

import {Title} from '../Title';

describe('@lad-tech/mobydick-core/Header', () => {
  it('renders default correctly', () => {
    const {toJSON} = render(<Title>My Text</Title>);
    expect(toJSON()).toMatchSnapshot();
  });
});
