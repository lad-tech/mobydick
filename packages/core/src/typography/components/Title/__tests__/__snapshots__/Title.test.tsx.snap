// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lad-tech/mobydick-core/Header renders default correctly 1`] = `
<Text
  allowFontScaling={true}
  style={
    [
      {
        "color": "#20242D",
        "fontFamily": "Inter",
        "fontSize": 36,
        "fontWeight": "600",
        "lineHeight": 40,
        "minHeight": 40,
      },
      undefined,
    ]
  }
>
  My Text
</Text>
`;
